x-pack/plugins/observability_solution/inventory/server/routes/entities/route.ts

```diff
@@ -100,7 +101,44 @@ export const listLatestEntitiesRoute = createInventoryServerRoute({
   },
 });
 
+export const groupEntitiesByRoute = createInventoryServerRoute({
+  endpoint: 'GET /internal/inventory/entities/group_by/{field}',
+  params: t.intersection([
+    t.type({ path: t.type({ field: t.string }) }),
```

Comment:
I'd say so, since I figure we don't want to group on just any kind of field, only those that we want to allow via the `Group By` selection button.

Do we have a list of fields we want to target as acceptable groupings?

Reply:
I'd say it should only accept `entity.type` for now, which is what the UI will currently accept. We can add more fields when necessary.

Reply:
In the long run, I think it won't matter. The user would be able to group by anything, and it'll be hard for us to keep track of everything.

Reply:
I think adding constraints helps align the expectations and document what the endpoint can handle. I imagine this endpoint will accept only `entity` fields in the group-by. If it's just a string, it means that we can group by any dymension.

Reply:
I restricted this to just `entity.type` for now. We can expand on this later.